src/main/java/com/pinterest/secor/common/LogFilePath.java

```diff
@@ -119,7 +119,7 @@ public LogFilePath(String prefix, String path) {
         // Parse basename.
         String basename = pathElements[pathElements.length - 1];
         // Remove extension.
-        int lastIndexOf = basename.lastIndexOf('.');
```

Comment:
What if the file name is a.b.c while a.b is the basename and c is the extension name?

Reply:
The logic to calculate the file extension name is in ParitionFinalizer line 65 (also in MessageWriter, line 64), I think we should use the same logic.  Can you move that logic into a common utility routine and have all three places calling that utility routine?

Reply:
> What if the file name is a.b.c while a.b is the basename and c is the extension name?
Yes, that would be a problem, but in that case we this constructor will fail.
we have this on line 131.
```java
        mGeneration = Integer.parseInt(basenameElements[0]);
```
Then we'd have `Integer.parse("a.b")`, so in that case this contructor would not support both, double extensions and files with multiple dots in their name. I made this change with asumption that base name of the file is allways a parseable number without dots.


> The logic to calculate the file extension name is in ParitionFinalizer line 65 (also in MessageWriter, line 64), I think we should use the same logic. Can you move that logic into a common utility routine and have all three places calling that utility routine?

You mean I should use it on `Uploader` class instead of using `srcPath.getExtention()`? My first idea was to do that, but since we already have extension on `srcPath`, and `dstPath` is path for trimmed srcFile, I thougt it is reasonable to reuse same extension instead regenerating it from configuration.
I do agree that PartitionFinalizer and MessageWritter logics should be extracted to common utility though :)

Reply:
I see.  In this case, can you rename the variable name 'lastIndexOf' to 'indexOf'?